src/Config.hs

```diff
@@ -25,99 +27,99 @@ instance Show StepAnswer where
 data PickInst = PickInst {
                  cnfs    :: ![Cnf]
                , correct :: !Int
-               , addText :: !(Maybe String)
+               , extraText :: Maybe (Map Language String)
```

Comment:
I changed it since all `Config` and `Inst` types in `Tasks` define it like this. Besides that, here is an idea: Why don't we just add the configurations to the respective instances? This would allow us to:

- Access the config in any grading scenario and apply feedback based on the config settings
- Refactor code so that we don't have to "forward" any config settings required to be available in the instance.
- (Avoid name overlaps)

Reply:
Okay, the point about some task types (in `Tasks` but not in `LogicTasks`) already doing it that way (with shared field names) had eluded me. It appears these were mostly (or all?) introduced by @owestphal quite recently (https://github.com/fmidue/logic-tasks/commit/30ae380cb3d6275a75f557eb0cf7b73b95a84924). I'm not sure it was intentional done that way, or violated some "policy" of `Config` and `Inst` types avoiding field overlap. My impression is that @marcellussiegburg usually avoids such name overlap. Maybe there are reasons "from the Autotool side".

About including `Config` types inside the `Inst` types: I'm almost certain this is a "no-go" from some Autotool usage perspective (for implementers and lecturers). The point being that in principle at least, an `Inst` should be able to be given ad-hoc, without knowledge about "how it came about from the config via a generator". One use case of this principle being that a lecturer should always be able to hand-craft an instance and provide that directly as an exercise task. (For demo purposes, to make sure that all students see the same instance because it has some interesting properties, etc.)

Reply:
>Maybe there are reasons "from the Autotool side".

No there are no such restrictions. But still I prefer having a single task type being unambiguous when imported into a single module. (Although there are ways around the issues, even in the future using e.g. qualified imports [i.e. migration strategy 1 in https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0366-no-ambiguous-field-access.rst])